src/bun.js/webcore/response.zig

```diff
@@ -1972,6 +1971,7 @@ pub const Fetch = struct {
         // used to clean up dynamically allocated memory on error (a poor man's errdefer)
         var is_error = false;
         var allocator = memory_reporter.wrap(bun.default_allocator);
+        errdefer bun.default_allocator.destroy(memory_reporter);
```

Comment:
This looks like a double free if `error.JSError` is returned because line 2039 sets `is_error` to true.
```zig
errdefer free(); // second free
defer {
    if (is_error) free();
}
defer {
    // ...
    if (!is_error and globalThis.hasException()) {
        is_error = true;
    }
}

global.throw();
return error.JSError;
```